api/preprints/serializers.py

```diff
@@ -559,12 +574,35 @@ def create(self, validated_data):
 
         title = validated_data.pop('title')
         description = validated_data.pop('description', '')
-        preprint = Preprint(provider=provider, title=title, creator=creator, description=description)
-        preprint.save()
+        preprint = Preprint.create(provider=provider, title=title, creator=creator, description=description)
 
         return self.update(preprint, validated_data)
 
 
+class PreprintCreateVersionSerializer(PreprintSerializer):
+    # Overrides PreprintSerializer to make id and title nullable
+    id = IDField(source='_id', required=False, allow_null=True)
+    title = ser.CharField(required=False)
+
+    create_from_guid = ser.CharField(required=True, write_only=True)
+
+    def create(self, validated_data):
+        create_from_guid = validated_data.pop('create_from_guid', None)
+        auth = get_user_auth(self.context['request'])
+        try:
+            preprint, update_data = Preprint.create_version(create_from_guid, auth)
+        except PermissionsError:
+            message = 'User must have admin permissions to create new version.'
+            sentry.log_message(message)
+            raise PermissionDenied(detail=message)
+        except UnpublishedPendingPreprintVersionExists:
+            message = 'Fail to create a new version since an unpublished pending version already exists.'
+            sentry.log_message(message)
+            raise Conflict(detail=message)
```

Comment:
No need to raise duplicate sentry messages that have been raised with more detail in `create_version()`. In addition, catch and handle `NotFound`.

Reply:
Fixed in https://github.com/CenterForOpenScience/osf.io/pull/10880